lib/compat/wordpress-6.1/blocks.php

```diff
@@ -392,3 +392,28 @@ function gutenberg_block_type_metadata_render_template( $settings, $metadata ) {
 	return $settings;
 }
 add_filter( 'block_type_metadata_settings', 'gutenberg_block_type_metadata_render_template', 10, 2 );
+
+/**
+ * Registers the metadata block attribute for block types.
+ *
+ * Once 6.1 is the minimum supported WordPress version for the Gutenberg
+ * plugin, this shim can be removed
+ *
+ * @param array $args Array of arguments for registering a block type.
+ * @return array $args
+ */
+function gutenberg_register_metadata_attribute( $args ) {
+	// Setup attributes if needed.
+	if ( ! isset( $args['attributes'] ) || ! is_array( $args['attributes'] ) ) {
```

Comment:
Why not use array_key_exists here too?

Reply:
Plus 1 to @scruffian 's nit.
```suggestion
	if ( ! array_key_exists( 'attributes', $args ) ) {
```

Reply:
`array_key_exists` will be truthy even if the var is `null`. I'm looking to determine that the value is completely "not a thing" rather than "the key exists".

Reply:
See prior art here

https://github.com/WordPress/gutenberg/blob/fedab3dc4250041f60a5068e683c5c617ac43786/lib/compat/wordpress-6.0/blocks.php#L182